tools/data-handler/src/update.ts

```diff
@@ -0,0 +1,132 @@
+/**
+    Cyberismo
+    Copyright © Cyberismo Ltd and contributors 2024
+
+    This program is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General Public License version 3 as published by the Free Software Foundation.
+
+    This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public
+    License along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+import { Project, ResourcesFrom } from './containers/project.js';
+import { Validate } from './validate.js';
+import {
+  resourceNameParts,
+  resourceNameToString,
+  ResourceName,
+} from './utils/resource-utils.js';
+import {
+  CardType,
+  FieldType,
+  LinkType,
+  Workflow,
+} from './interfaces/resource-interfaces.js';
+
+/**
+ * Class that handles 'update' commands.
+ */
+export class Update {
+  constructor(private project: Project) {}
+
+  // Does the actual change to a value.
+  private async doUpdateResource(
+    resourceName: ResourceName,
+    key: string,
+    value: unknown,
+  ) {
+    const resource = await this.resource(resourceName);
+    if (resource && resource.name) {
+      if (key === 'name') {
+        resource.name = value as string;
+      } else if (key === 'workflow') {
+        (resource as CardType).workflow = value as string;
```

Comment:
Could consider making a typescript type, which maps ResourceName to a a value type. Then this function could be generic

Reply:
Can you expand on that? `ResourceName` is just `name` (prefix/type/identifier) and `path`

Reply:
I thought something like this would work, but looks like the compiler did not manage to figure out the type in the if block:
```
type Resource = 'workflow' | 'fieldtype';

export interface ResourceName<T extends Resource> {
  prefix: string;
  type: T;
  identifier: string;
}

export type ValueMap = {
  workflow: string;
  fieldtype: number;
};

function func<T extends Resource>(name: ResourceName<T>, value: ValueMap[T]) {
  if (name.type === 'fieldtype') {
    // value is now number
  } else {
    // value is now string
  }
}
```